python/utils/OpaqueArguments.h

```
@@ -274,6 +273,92 @@ inline void handleStructMemberVariable(void *data, std::size_t offset,
       });
 }
 
+/// @brief For the current vector element type, insert the
+/// value into the dynamically-constructed vector.
+inline void handleVectorElements(void *data, Type eleTy, py::list list) {
+  auto appendValue = []<typename T>(py::list list, void *data,
+                                    auto &&converter) {
+    std::vector<T> *values = new std::vector<T>(list.size());
+    for (std::size_t i = 0; auto &v : list) {
+      auto converted = converter(v, i);
+      (*values)[i++] = converted;
+    }
+    std::memcpy(((char *)data), values, sizeof(std::vector<T>));
```

This must leak memory since `values` silently falls out of scope. The `new vector` is creating a vector object (say at memory address _A_) and an associated buffer that can hold `list.size()` elements (say at address _B_) and that buffer is initialized to the default value of `T` (aka, `T{}`). So after the new `values` contains the address _A_ and the pointers at _A_ point to _B_.

The `memcpy` is bitwise copying the vector object. So the pointers at _A_ and copied to `data`. At this point there are two vector objects with identical data and assumed ownership.

When this function exits, `values` still points to _A_ and there is nothing to delete _A_ from the heap. Which is semi-good, because the `vector::~vector` would, by the rule of ownership, free everything at _B_, which would mean `data` would have pointers to freed memory. Instead of that bug though, we are leaking _A_.

---

I tried to fix it. Let me know what you think.